### ISSUE TITLE
[Bug] v.neighbors crash when region only subset of vector map

### ISSUE THREAD

**Issue by mlennert** [OSGeo/grass#535]
NB: I discovered this issue while working on #534.

Here's the issue:

```g.region rast=elevation
v.neighbors input=schools_wake output=schools_count size=3000 --o
ATTENTION: Input vector and computational region do not overlap
ERREUR : Unable to seek: Mauvais descripteur de fichier
```

Here's the end of the DEBUG=2 output:

```D2/3: G_file_name(): path = /home/mlennert/GRASSDATA/nc_spm_08_grass7/PERMANENT/vector/schools_wake/hist
D2/3: 	file open: read (mode = r)
D1/3: Vect_Rewind(): name = schools_wake level = 2
D1/3: Vect_get_field_number(): field = -1
D1/3: Vect_close(): name = schools_wake, mapset = PERMANENT, format = 0, level = 2, is_tmp = 0
D1/3: spatial index file closed
D1/3: close history file
D1/3: V1_close_nat(): name = schools_wake mapset= PERMANENT
ATTENTION: Input vector and computational region do not overlap
D2/3: G_file_name(): path = /home/mlennert/GRASSDATA/nc_spm_08_grass7/user1/GDAL
D2/3: G_file_name(): path = /home/mlennert/GRASSDATA/nc_spm_08_grass7/user1
D2/3: G__temp_element(): .tmp/bonheur (tmp=0)
D2/3: G_file_name(): path = /home/mlennert/GRASSDATA/nc_spm_08_grass7/user1/.tmp/bonheur/30662.4
D2/3: G_tempfile_pid(): /home/mlennert/GRASSDATA/nc_spm_08_grass7/user1/.tmp/bonheur/30662.4
D2/3: G_file_name(): path = /home/mlennert/GRASSDATA/nc_spm_08_grass7/user1
D2/3: G_file_name(): path = /home/mlennert/GRASSDATA/nc_spm_08_grass7/user1
D2/3: G__temp_element(): .tmp/bonheur (tmp=0)
D2/3: G_file_name(): path = /home/mlennert/GRASSDATA/nc_spm_08_grass7/user1/.tmp/bonheur/30662.5
D2/3: G_tempfile_pid(): /home/mlennert/GRASSDATA/nc_spm_08_grass7/user1/.tmp/bonheur/30662.5
D3/3: Vect_select_lines_by_box()
D3/3:   Box(N,S,E,W,T,B): 2.299950e+05, 2.269950e+05, 6.315050e+05, 6.285050e+05, 1.797693e+308, -1.797693e+308
D3/3: dig_select_lines_with_box()
ERREUR : Unable to seek: Mauvais descripteur de fichier
D1/3: G_set_program_name(): g.gisenv
D3/3: G_option_to_separator(): key = separator -> sep = '/'
```

So the error happens in Vect_select_lines_by_box() when calling dig_select_lines(), so apparently linked to the spatial index. 

@metzm could you have a look at this ?

**Comment by mlennert** [comment]
Found it myself. The problem was actually in the module itself: 
In [lines 95-99](https://github.com/OSGeo/grass/blob/12ad52323315cbd7aa69c15a1f0f7c2527805c2e/vector/v.neighbors/main.c#L95-L99) the input vector map is closed, but only a warning is emitted, then later the module tries to access the same vector map again.
I don't think that closing this map makes any sense and the module seems to run fine without doing so, so I'll take that out. I'll integrate this change into PR #534

**Comment by mlennert** [comment]
Fixed in #534
